core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -111,8 +113,14 @@ class TaskMetrics private[spark] () extends Serializable {
    * across all such data structures created in this task. For SQL jobs, this only tracks all
    * unsafe operators and ExternalSort.
    */
+  // TODO: SPARK-48789: the naming is confusing since this does not really reflect the whole
```

Comment:
Given that `TaskMetrics` is marked as `@DeveloperAPI`, I think we should also add Scaladoc API docs for the two new `Peak*ExecutionMemory` metrics. We might also want to explicitly call out in the `peakExecutionMemory` API doc that it is _not_ equal to the sum of the other two metrics.

Reply:
Updated